docs/public-api.md

```diff
@@ -0,0 +1,65 @@
+# SwiftNIO 2 Public API
+
+The SwiftNIO project (which includes all of `github.com/apple/swift-nio*`)aims to follow [SemVer 2.0.0](https://semver.org/spec/v2.0.0.html) which requires to declare a public API. This document along with the [API documentation](https://apple.github.io/swift-nio/) declare SwiftNIO's API.
+
+### What Are Acceptable Uses of NIO's Public API
```

Comment:
the `- ` formatting is a bit hard to read to be honest; Since the new point is very close to the end of the examples/reasoning of previous one; Something to separate would be useful, be it a rule number or a --- ?


Or like:

#### Extensions to NIO types (level 3 or level 4)
what is allowed... 
Why? 
examples